### PR TITLE
add_docker_image_to_jenkinsfile

### PR DESCRIPTION
**Pull request by MichaelTzach**
added docker image to npm install stage and moved npm install to run on non master branches

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -5,16 +5,22 @@ pipeline {
   agent any
 
   stages {
+    stage('Build') {
+      script {
+        // package.json should not have private dependencies
+        // we should be able to install dependencies without credentials
+        docker.image('node:12-alpine').inside() {
```
Comment (yosiat) [inline Jenkinsfile]: why not use Dockerfile.ci file like other projects have?

Comment (yosiat) [inline Jenkinsfile]:
```suggestion
        docker.image('bringg/node:12-alpine').inside() {
```

Comment (MichaelTzach) [reply Jenkinsfile]: done

Comment (MichaelTzach) [reply Jenkinsfile]: discussed offline. will leave it as is.

**Pull request merged**